resources/js/src/app/mixins/baseDropdown.js

```diff
@@ -0,0 +1,36 @@
+export default {
+    props: [
+        "template",
+        "facet"
+    ],
+
+    data: function () {
+        return {
+            isShowDropdown: false
+        }
+    },
+
+    created() {
+        this.$options.template = this.template || "#vue-item-dropdown";
+    },
+
+    computed: {
+        ...Vuex.mapState({
+            isLoading: state => state.itemList.isLoading
+        })
+    },
+
+    methods: {
+        selected: function (value) {
+            this.updateSelectedFilters(this.facet.id, value);
+        },
+
+        hideDropdown: function () {
+            this.isShowDropdown = false;
+        },
+
+        toggleDropdown: function () {
+            this.isShowDropdown = !this.isShowDropdown;
+        }
+    }
+};
```

Comment:
Please add a newline to the end of the file. Hopefully we have a linter for JS files as well one day ;).